### PR TITLE
Integrate orderlist component for experiences

### PR DESCRIPTION
**Pull request by Rahul139-stack**
**Sample**:
image

### CODE REVIEW

File: src/components/Dialog/ExperiencePopup.vue [hunk 1]
```diff
@@ -0,0 +1,189 @@
+<template>
+    <div class="experienceDialog">
+        <Dialog v-model:visible="displayEditPopup" :header="$t(`components.dialog.popup.heading`)" :modal="true"
+            :closable="false" :style="{ width: '350px' }">
+            <template #header>
+                <!-- add here header -->
```
Comment (Harikarthyk) [inline src/components/Dialog/ExperiencePopup.vue:6]: remove comments if not required

File: src/components/Dialog/ExperiencePopup.vue [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        updateExperinceInfo: { type: Function, default() { return {}} }
+    },
+    emits: ["close"],
+    mounted() {
+        this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
+        if (this.type === "UPDATE") {
+            this.label = this.payload.label,
+                this.icon = this.payload.icon,
+                this.url = this.payload.url,
+                this.editElements = true
```
Comment (Harikarthyk) [inline src/components/Dialog/ExperiencePopup.vue:105]: align code properly.

File: src/components/Dialog/ExperiencePopup.vue [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+.deleteIcon {
+    font-size: 2rem;
+}
+
+.p-dialog-header {
+      padding: 1.25rem 1.8rem;
+
+      .p-dialog-header-icon:last-child {
+     margin-left:10rem;
```
Comment (Harikarthyk) [inline src/components/Dialog/ExperiencePopup.vue:180]: Format the code.

File: src/components/Settings/General.vue [hunk 4]
```diff
@@ -1,56 +1,155 @@
...
+    saveOrderList() {
+      settingsModule.updateList(this.updatedExperienceList)
+      this.saveDiabale = true
+      this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
+
+    },
+    handleFeatureToggle(enabled, id) {
+      // alert("hii",id,enabled)
+      console.log("hello")
+      console.log("id-ena", id, enabled)
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:100]: remove console added for testing.

File: src/components/Settings/General.vue [hunk 5]
```diff
@@ -1,56 +1,155 @@
...
+      this.listOfExperiences = this.listOfExperiences.map(item => {
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:101]: If you are fetching the this.listOfExperiences from the store then what is the purpose of this ?

File: src/components/Settings/General.vue [hunk 6]
```diff
@@ -1,56 +1,155 @@
...
+    updateExperinceInfo(id,newPayload){
+      this.listOfExperiences = this.listOfExperiences.map(item => {
+        console.log("item",item)
+        if (item.id === id) {
+          item=newPayload
+        }
+        console.log("item",item)
+        return item;
+      });
+      this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:120]: same here.

File: src/components/Settings/General.vue [hunk 7]
```diff
@@ -1,56 +1,155 @@
...
+    },
+    
+    editDialogToggle(item) {
+      console.log("item", item)
+      this.editModalPayload = item
+      this.type = "UPDATE"
+      this.editModalIsOpen = true
+    },
+    editClosemodal() {
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:129]: Please follow camelCase

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: src/components/Dialog/ExperiencePopup.vue (hunk 1 above)
Comment (Rahul139-stack) [reply src/components/Dialog/ExperiencePopup.vue:6]: Removed

File: src/components/Dialog/ExperiencePopup.vue (hunk 2 above)
Comment (Rahul139-stack) [reply src/components/Dialog/ExperiencePopup.vue:105]: Aligned

File: src/components/Dialog/ExperiencePopup.vue (hunk 3 above)
Comment (Rahul139-stack) [reply src/components/Dialog/ExperiencePopup.vue:180]: Formatted

File: src/components/Settings/General.vue (hunk 4 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue:100]: removed

File: src/components/Settings/General.vue (hunk 7 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue:129]: Done

File: src/components/Settings/General.vue (hunk 5 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue]: Refactored code

File: src/components/Settings/General.vue (hunk 6 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue:94]: Refactored code

File: src/components/IframeComponent/index.vue [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const settingsModule = getModule(Settings)
+export default {
+  name: "IframeComponent",
+  components: { MenuBar, MainMenu, Loader },
+  data() {
+    return {
+      showMainOverlayMenu: false,
+      mainMenuVisible: true,
+      Experience: {},
+      loading: true
```
Comment (Harikarthyk) [inline src/components/IframeComponent/index.vue:33]: Please follow camelCase standards for the variable name.

File: src/components/IframeComponent/index.vue [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    <loader />
+  </div>
+  <div class="app">
+    <div class="header" :class="{ 'menu-hidden': !mainMenuVisible }">
+      <main-menu :main-menu-visible="mainMenuVisible" @toggle="toggleMainMenu" :subMenuVisible="false" />
+      <menu-bar :menu-bar-visible="mainMenuVisible" @toggle="toggleMainMenu" />
+    </div>
+    <!-- content -->
+    <div class="iframe-container">
+      <object :data="`${Experience.url}`" width="100%" height="100%" @load="onObjLoad()"></object>
```
Comment (Harikarthyk) [inline src/components/IframeComponent/index.vue:12]: Please follow camelCase standards for the variable name.

File: src/components/Settings/General.vue [hunk 10]
```diff
@@ -1,56 +1,129 @@
...
-        @input="handleFeatureToggle($event, feature.id)"
-      />
-      <label for="username">{{
-        $t(`components.setting.general.${feature.id}`)
-      }}</label>
+    <div class="topButtons">
+      <Button class="p-button-raised mr-2" :label="$t(`components.setting.general.buttons.addlabel`)"
+        @click="addDialogToggle()" ></Button>
+      <Button class="p-button-raised mr-2" :label="$t(`components.setting.general.buttons.editlabel`)"
+        :disabled="saveDiabale" @click="saveOrderList()" ></Button>
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:7]: Please correct the typo

File: src/components/Settings/General.vue [hunk 11]
```diff
@@ -1,56 +1,129 @@
...
+    editDialog() {
+      return settingsModule.data.editDialog
     },
-    methods: {
-      handleFeatureToggle(enabled, id) {
-        settingsModule.toggleExperimentalFeatures({ id, enabled })
-      },
+  },
+  mounted() {
+    this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
```
Comment (Harikarthyk) [inline src/components/Settings/General.vue:75]: you can use the function defined below for this.

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: src/components/IframeComponent/index.vue (hunk 9 above)
Comment (Rahul139-stack) [reply src/components/IframeComponent/index.vue:12]: Done

File: src/components/Settings/General.vue (hunk 10 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue:7]: Done

File: src/components/Settings/General.vue (hunk 11 above)
Comment (Rahul139-stack) [reply src/components/Settings/General.vue:75]: Done

File: src/components/Settings/General.vue [hunk 12]
```diff
@@ -62,7 +62,7 @@ export default {
       editModalPayload: {},
       editModalIsOpen: false,
       type: "",
-      saveDiabale: true
+      saveDiable: true
```
[the remark this answers is not in the document]
Comment (Rahul139-stack) [reply src/components/Settings/General.vue]: Renamed

**Review by Harikarthyk — APPROVED** [review]

**Pull request merged**